app/models/concerns/status_threading_concern.rb

```diff
@@ -86,6 +86,9 @@ def find_statuses_from_tree_path(ids, account)
 
     # Order ancestors/descendants by tree path
     statuses.sort_by! { |status| ids.index(status.id) }
+
+    # Bring self-replies to the top
+    statuses.sort_by! { |status| status.in_reply_to_account_id == status.account_id ? -1 : 0 }
```

Comment:
How reliable is that (is there any guarantee to respect the previous order)? Also, what happens for self-replies after someone else replied? (e.g, A replies to themselves in a reply to B, which replied to the original post from A)